public/app/features/transformers/partitionByValues/partitionByValues.ts

```diff
@@ -0,0 +1,107 @@
+import { map } from 'rxjs';
+
+import {
+  ArrayVector,
+  DataFrame,
+  DataTransformerID,
+  SynchronousDataTransformerInfo,
+  getFieldMatcher,
+  RegexpOrNamesMatcherOptions,
+} from '@grafana/data';
+import { getMatcherConfig } from '@grafana/data/src/transformations/transformers/filterByName';
+import { noopTransformer } from '@grafana/data/src/transformations/transformers/noop';
+
+import { partition } from './partition';
+
+export interface FrameNameOptions {
+  /** if false, will only include values without key field names, e.g. 'Europe Chef' */
+  names?: boolean; // false
+  /** name/value separator, e.g. '=' in 'Region=Europe' */
+  separator1?: string;
+  /** name/value pair separator, e.g. ' ' in 'Region=Europe Profession=Chef' */
+  separator2?: string;
+}
```

Comment:
these are not currently exposed in the transformer UI, but allow you to choose how the resulting frames are named based on the found fields' unique values